src/common/types.ts

```diff
@@ -110,16 +115,15 @@ export type RouterStoreContext = {
   location: Location;
 };
 
-export type Route = {
+/**
+ * Invariant route
+ *
+ * Base type for route, which doesn't contain implementation details
+ */
+export type InvariantRoute = {
   path: string;
   exact?: boolean;
 
-  /** The component to render on match, typed explicitly */
-  component: ComponentType<RouteContext>;
-
-  /** If present, must return true to include the route. */
-  enabled?: () => boolean;
-
   /** Signals that this is a redirect route, we need to handle these in a special way. */
   isRedirect?: boolean;
```

Comment:
nit; this isn't being used any more. Would you mind removing it?

Reply:
Done 👍